src/Connection/CountLimitingConnectionPool.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+namespace Amp\Http\Client\Connection;
+
+use Amp\CancellationToken;
+use Amp\Deferred;
+use Amp\Http\Client\Internal\ForbidSerialization;
+use Amp\Http\Client\Request;
+use Amp\Http\Client\Response;
+use Amp\MultiReasonException;
+use Amp\Promise;
+use Amp\Success;
+use function Amp\call;
+use function Amp\coroutine;
+
+final class CountLimitingConnectionPool implements ConnectionPool
```

Comment:
Should this be `ConnectionLimitingConnectionPool` (vs. `StreamLimitingConnectionPool`)? I guess that's a stupid name.

Reply:
I thought of that name too, but yes, it's a stupid name. `ConnectionCountLimitingPool`?

Reply:
And maybe `StreamCountLimitingPool`?

Reply:
Maybe just `StreamLimitingPool` and `ConnectionLimitingPool`?

Reply:
Works for me. Done.